packages/components/src/components/hive-navigation/index.tsx

```diff
@@ -0,0 +1,562 @@
+import React, { forwardRef, Fragment, ReactNode, useEffect, useRef } from 'react';
+import { createPortal } from 'react-dom';
+import { useRouter } from 'next/router';
+import { useMenu, useThemeConfig } from 'nextra-theme-docs';
+import { useMounted } from 'nextra/hooks';
+import { MenuIcon } from 'nextra/icons';
+import { cn } from '../../cn';
+import { renderSlot } from '../../helpers/render-slot';
+import { GraphQLFoundationLogo, GuildLogo, HiveCombinationMark, TheGuild } from '../../logos';
+import { PRODUCTS, SIX_HIGHLIGHTED_PRODUCTS } from '../../products';
+import { Anchor } from '../anchor';
+import { CallToAction } from '../call-to-action';
+import {
+  AccountBox,
+  AppsIcon,
+  ArrowIcon,
+  BardIcon,
+  GitHubIcon,
+  GroupIcon,
+  HiveIcon,
+  HonourIcon,
+  PaperIcon,
+  PencilIcon,
+  RightCornerIcon,
+  ShieldFlashIcon,
+  TargetIcon,
+} from '../icons';
+import {
+  NavigationMenu,
+  NavigationMenuContent,
+  NavigationMenuItem,
+  NavigationMenuLink,
+  NavigationMenuList,
+  NavigationMenuTrigger,
+} from './navigation-menu';
+
+export * from './graphql-conf-card';
+
+const EXPLORE_HREF = 'https://github.com/the-guild-org';
+
+const ENTERPRISE_MENU_HIDDEN = true;
+
+export interface HiveNavigationProps {
+  companyMenuChildren?: ReactNode;
+  children?: ReactNode;
+  className?: string;
+  /**
+   * We change links to relative based on what product we're in.
+   */
+  productName: string;
+}
+/**
+ *
+ * @example
+ * ```tsx
+ * <HiveNavigation
+ *   companyMenuChildren={<GraphQLConfCard image={graphQLConfLocalImage} />}
+ *   items={items}
+ * >
+ *   {extraContent}
+ * </HiveNavigation>
+ * ```
+ */
+export function HiveNavigation({
+  companyMenuChildren,
+  children,
+  className,
+  productName,
+}: HiveNavigationProps) {
+  // `useThemeConfig` doesn't return anything outside of Nextra, and the provider isn't exported
+  const themeConfig = useThemeConfig() as ReturnType<typeof useThemeConfig> | undefined;
+  const Search = themeConfig?.search?.component;
+
+  const isHive = productName === 'Hive';
+
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  return (
+    <div
+      ref={containerRef}
+      className={cn(
+        'sticky top-0 z-20 border-b border-beige-400/[var(--border-opacity)] bg-[rgb(var(--nextra-bg))] px-6 py-4 text-green-1000 transition-[border-color] duration-500 md:mb-[7px] md:mt-2 dark:border-neutral-700/[var(--border-opacity)] dark:text-neutral-200 [&.light]:border-beige-400/[var(--border-opacity)] [&.light]:bg-white [&.light]:text-green-1000',
+        className?.includes('light') && 'light',
+      )}
+      style={{ '--border-opacity': 0 }}
+    >
+      <TopOfSiteMarker
+        onChange={scrolled => {
+          const container = containerRef.current;
+          if (container) {
+            container.style.setProperty('--border-opacity', scrolled ? '1' : '0');
+          }
```

Comment:
```suggestion
            container.style.setProperty('--border-opacity', scrolled ? '1' : '0');
```

Reply:
Once the component unmounts, React will set ref.current to `null`, so this could crash during unmount.

Reply:
Okay, it feels scary, but I guess this component won't ever get unmounted, as it's on the top of every page, and even if it did crash, it's just a handler, so worst case it logs an error to the console.